From 7b8003e341c0ebbf4cf097799e1223d6017dd15e Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 05 Jun 2019 13:21:17 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: let NIC2_HBM pll gated w/a

The NIC2_HBM PLL HBW output cannot be gated, but other (e.g. LBW)
outputs, might be able to be gated for the W/A.
So let the hl_pll_gated_clk_sw() function decide instead of disabling
this completely for all NIC2_HBM PLL outputs.

Change-Id: Ib8ee35d2b4e585784c2f043df089f230c0a358a7
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 087edf4..2e254a0 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -358,6 +358,11 @@
 	u32 gmask = hl_pll_build_gate_mask(cfg);
 	u64 reg;
 
+	if (gmask == 0xF) {
+		preboot_puts("Gate mask is 0xF - no gate, skipping...\n");
+		return;
+	}
+
 	regmap_writel(gmask, cfg->base + HL_PLL_CLK_GATER);
 	hl_pll_udelay(1);
 
@@ -390,7 +395,7 @@
 				preboot_puts("NIC2_HBM configuration done, resetting...\n");
 				gaudi_reset_soc();
 			}
-			continue;
+			break;
 		default:
 			break;
 		}
